### ISSUE TITLE
About pytorch average mIOU(over 6 areas) for S3DIS.

### ISSUE THREAD

**Issue by a18700** [lightaime/deep_gcns_torch#17]
hello authors,

we want to get overall mIOU(6 areas) for my own trained models, but how can we get it in pytorch?

is it enough for altering from 5 to 1, ..., 6 of below sentence of attaching code(line 17-25 of test.py),
'test_dataset = GeoData.S3DIS(opt.test_path, 5, False, pre_transform=T.NormalizeScale())' ?

======
def main():
opt = OptInit().initialize()

print('===> Creating dataloader...')
test_dataset = GeoData.S3DIS(opt.test_path, 5, False, pre_transform=T.NormalizeScale())
test_loader = DenseDataLoader(test_dataset, batch_size=opt.batch_size, shuffle=False, num_workers=0)
opt.n_classes = test_loader.dataset.num_classes
if opt.no_clutter:
    opt.n_classes -= 1
======

If the fast reply possible, we will be really grateful of you.

Thank you!

**Comment by lightaime** [comment]
Yes. 5 is the leave-out area here. For example, if you want to train on areas 2-6 and test on area 1. You can set it as 1 for both the train dataloader and test dataloader.

**Comment by a18700** [comment]
> Yes. 5 is the leave-out area here. For example, if you want to train on areas 2-6 and test on area 1. You can set it as 1 for both the train dataloader and test dataloader.

thanks for quick reply.

so to get average 6 area miou, (i want to get number, 60% miou for baseline model in 6 areas) is from an average six cases, which 1st case : train on 2-6 and test on 1, ..., train on 1-5 and test on 5? it's look strange..

I think it's not possible to generate 6 average mIOU results by directly using pytorch code, because currently train loader exclude training samples of area 5, am i right? :)

**Comment by lightaime** [comment]
People usually report the mIOU by leave-one-out strategy which means we
train on areas 2-6 and test on area 1, train on 1,3-6 and test on 2, train
on 1-2,4-6 and test on 3... train on 1-5, test on 6. In order to generate
mIOU for 6 areas, you need to train 6 times. And area 5 is the area that
usually people do ablation study on. So here our example is trained on
1-4,6 and tested 5.

**Comment by lightaime** [comment]
I close this issue since it seems to be resovled.
